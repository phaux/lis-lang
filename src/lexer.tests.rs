use super::*;

#[test]
fn prog_simple() {
    let input = "let x = 5;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_escaped() {
    let input = r#" s = "hello\nworld\twith\\quotes\"" "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_str_unclosed() {
    let input = r#" s = "hello "#;
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num() {
    let input = " x = 5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn assign_num_invalid() {
    let input = " x = 5.5.5 ";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_line() {
    let input = "let x = 1; // this is a comment\nprint x";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}

#[test]
fn comment_block() {
    let input = "let x = /* block comment */ 1;";
    let tokens: Vec<_> = Lexer::new(input).collect();
    insta::assert_debug_snapshot!(tokens);
}
