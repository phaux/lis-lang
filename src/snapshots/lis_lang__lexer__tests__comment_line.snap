---
source: src/lexer.tests.rs
expression: tokens
---
[
    Token {
        sigil: Keyword(
            Let,
        ),
        range: 0..3,
    },
    Token {
        sigil: Ident {
            name: "x",
        },
        range: 4..5,
    },
    Token {
        sigil: Eq,
        range: 6..7,
    },
    Token {
        sigil: Num {
            val: 1.0,
        },
        range: 8..9,
    },
    Token {
        sigil: Semi,
        range: 9..10,
    },
    Token {
        sigil: Comment {
            body: " this is a comment",
        },
        range: 11..31,
    },
    Token {
        sigil: Keyword(
            Print,
        ),
        range: 32..37,
    },
    Token {
        sigil: Ident {
            name: "x",
        },
        range: 38..39,
    },
]
